library/spdm_requester_lib/libspdm_req_get_key_pair_info.c

```diff
@@ -167,7 +178,40 @@ static libspdm_return_t libspdm_try_get_key_pair_info(libspdm_context_t *spdm_co
         status = LIBSPDM_STATUS_INVALID_MSG_SIZE;
         goto receive_done;
     }
-    spdm_response_size = sizeof(spdm_key_pair_info_response_t) + spdm_response->public_key_info_len;
+
+    rsp_pqc_asym_algo_capabilities = 0;
+    rsp_current_pqc_asym_algo = 0;
+    if (spdm_response->header.spdm_version >= SPDM_MESSAGE_VERSION_14) {
+        if (spdm_response_size < sizeof(spdm_key_pair_info_response_t) +
+            spdm_response->public_key_info_len + sizeof(uint32_t)) {
+            status = LIBSPDM_STATUS_INVALID_MSG_SIZE;
+            goto receive_done;
+        }
+        ptr = (uint8_t *)spdm_response + sizeof(spdm_key_pair_info_response_t) + spdm_response->public_key_info_len;
+        pqc_asym_algo_cap_len = libspdm_read_uint32 (ptr);
+        if (spdm_response_size < sizeof(spdm_key_pair_info_response_t) +
+            spdm_response->public_key_info_len + sizeof(uint32_t) + pqc_asym_algo_cap_len + sizeof(uint32_t)) {
+            status = LIBSPDM_STATUS_INVALID_MSG_SIZE;
+            goto receive_done;
+        }
+        current_pqc_asym_algo_len = libspdm_read_uint32 (ptr + sizeof(uint32_t) + pqc_asym_algo_cap_len);
+
+        if (current_pqc_asym_algo_len > sizeof(uint32_t)) {
+            current_pqc_asym_algo_len = sizeof(uint32_t);
+        }
+        libspdm_copy_mem (&rsp_current_pqc_asym_algo, sizeof(rsp_current_pqc_asym_algo),
+                          ptr + sizeof(uint32_t) + pqc_asym_algo_cap_len + sizeof(uint32_t), current_pqc_asym_algo_len);
+        if (pqc_asym_algo_cap_len > sizeof(uint32_t)) {
```

Comment:
For 1.4 only need to read the first two bytes of `PqcAsymAlgoCapabilities`.

Reply:
Yes, 2 bytes is enough. But I want to align to base_pqc_asym, which is uint32_t.

Also, we still support the peer to use 2 bytes.